com.unity.netcode.gameobjects/TestHelpers/Runtime/NetcodeIntegrationTest.cs

```diff
@@ -79,6 +79,15 @@ public static void DeregisterNetworkObject(ulong localClientId, ulong networkObj
         protected const uint k_DefaultTickRate = 30;
         protected abstract int NumberOfClients { get; }
 
+        /// <summary>
+        /// Set this to false to create the clients first.
+        /// Note: If you are using scene placed NetworkObjects or doing any form of scene testing and
+        /// get prefab hash id "soft synchronization" errors, then set this to false and run your test
+        /// again.  This is a work-around until we can resolve some issues with NetworkManagerOwner and
+        /// NetworkManager.Singleton.
+        /// </summary>
```

Comment:
do you mind elaborating a little more? what's going on here?

Reply:
Its a little complicated, but it has to do with our integration tests running in the same thread, how we populate newly loaded (not yet "spawned") in-scene placed NetworkObjects, and NetworkObject.NetworkManager referencing NetworkManager.Singleton if NetworkObject.NetworkManagerOwner is null.

- Same Thread: In several places we check to make sure the NetworkObject.NetworkManager is the "right one" for the NetworkManager context
- Populating In-Scene Objects:  This checks to make sure the NetworkSceneManager instance is the same as the NetworkObject.NetworkManager before adding it to the ScenePlacedObjects list.
- Default NetworkManager:  This happens to be NetworkManager.Singleton that is set upon the first NetworkManager being instantiated.

So, under some conditions you might want NetworkManager.Singleton to be the server and under other conditions you might want it to be a client.  If you are loading a scene with in-scene placed NetworkObjects, then you will want the NetworkManager.Singleton to be the first client instantiated in order to properly synchronize/find the in-scene placed NetworkObjects (which are loaded by the server and not loaded by the client since they share the same scene space).

If you want more details feel free to DM me in slack... really we need to handle in-scene placed NetworkObjects better... like that "pre-rfc" document outlines.